File: object_Movie/src/main/java/objectstudy/policy/refund/AmountRefundPolicy.java
package objectstudy.policy.refund;

import objectstudy.Money;
import objectstudy.Refund;
import objectstudy.Reservation;
import objectstudy.condition.refund.RefundCondition;

public class AmountRefundPolicy extends DefaultRefundPolicy{

    public AmountRefundPolicy(RefundCondition... conditions) {
        super(conditions);
    }

    @Override
    protected Money getDiscountAmount(Reservation reservation) {

        return reservation.calculateRefundFee();
    }
}

File: object_Movie/src/main/java_to_kotlin/policy/refund/AmountRefundPolicy.kt
package java_to_kotlin.policy.refund

import java_to_kotlin.Money
import java_to_kotlin.Refund
import java_to_kotlin.condition.refund.RefundCondition

class AmountRefundPolicy(vararg conditions: RefundCondition?) : DefaultRefundPolicy(*conditions) {
    override fun getDiscountAmount(refund: Refund): Money? {
        return refund.movieFee
    }
}
